From 4aa78f52232682b353eb15c219171e466987bac7 Mon Sep 17 00:00:00 2001
From: Maxim Cournoyer <maxim.cournoyer@gmail.com>
Date: Thu, 16 Jan 2020 00:19:29 -0500
Subject: [PATCH] build: Enable assertions for the RelWithDebugInfo build type.

This fixes issue #5 (see:
https://gitlab.com/inkscape/lib2geom/issues/5).

* CMakeLists.txt: Remove the "-DNDEBUG" CXX flag from the default
configuration for the RelWithDebugInfo build type.
---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bfeb8f03..a663a1b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,14 @@ enable_testing()
 
 include(CheckCXXSourceCompiles)
 
+# Enable assertions for the RelWithDebugInfo build type.  This is
+# useful as some tests make use of it (see:
+# https://gitlab.com/inkscape/lib2geom/issues/5).
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  string(REPLACE "-DNDEBUG" ""
+    CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
+endif()
+
 # Find dependencies
 find_package(Boost 1.40 REQUIRED)
 find_package(DoubleConversion REQUIRED)
-- 
2.24.1

